File: api/entities/Item.go
package entities

type Item struct {
	Id          int          `json:"id"`
	Name        string       `json:"name"`
	Recipe      []Ingredient `json:"recipe"`
	Time        float32      `json:"time"`
	Result      int          `json:"result"`
	MachineType string       `json:"machineType"`
}

File: api_client/src/components/ui/Item.js
/* eslint-disable no-unreachable */
import React from "react";
import "./item.css";
import Recipe from "./Recipe";

const Item = ({item}) => {
  console.log(item)
  return (
    <div data-cy="article-container" className="container">
      <div data-cy="item" className="itemContainer">
        <p data-cy="name" className="title">{item.name}</p>
        {item.time !== 0 && <p data-cy="name" className="content">Crafting time: {item.time}</p>}
        <p data-cy="name" className="content">Machine type: {item.machineType}</p>
        {item.recipe && <Recipe ingredients={item.recipe}/>}
      </div>
    </div>
  );
};

export default Item;